### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by xaque208**

### CODE REVIEW

File: lib/puppet_x/ldapquery.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+      begin
+        ldap.search(:base => base,
+                    :filter => ldapfilter,
+                    :attributes => attributes,
+                    :time => 10) do |entry|
+          entries << entry
+        end
+        Puppet.debug(entries)
+        File.open('/Users/zach/Code/modules/ldapquery/entries.obj', 'w') { |file| file.write(Marshal.dump(entries)) }
```
Comment (danieldreier) [inline lib/puppet_x/ldapquery.rb]: think you left a hardcoded path here?

File: spec/fixtures/entries_multivalue.obj [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Iu:Net::LDAP::Entry�dn: uid=zach,ou=users,dc=puppetlabs,dc=com
```
Comment (danieldreier) [inline spec/fixtures/entries_multivalue.obj]: I'm seeing a weird character here and in the first line of `spec/fixtures/entries_objectClass.obj` - encoding issue?

File: spec/functions/ldapquery_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  #subject do
+  #  function_name = Puppet::Parser::Functions.function(:ldapquery)
+  #  scope.method(function_name)
+  #end
+
+  #good_params = '("uid=zach")'
+  #expected = {}
+
+  #it { should run.with_params(good_params).and_return(expected) }
```
Comment (danieldreier) [inline spec/functions/ldapquery_spec.rb]: are the tests supposed to be commented out?

File: spec/unit/ldapquery_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    "shadowAccount",
+                    "inetOrgPerson",
+                    "puppetPerson",
+                    "ldapPublicKey",
+                    "top"]}]
+
+        entries = Marshal.load(File.read("spec/fixtures/entries_objectClass.obj"))
+
+        l = PuppetX::LDAPquery.new(filter, attributes, base)
+        #expect(l).to receive(:get_config).and_return(real_conf)
```
Comment (danieldreier) [inline spec/unit/ldapquery_spec.rb]: dead code?

File: spec/unit/ldapquery_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        wanted = [{"dn"=>"uid=zach,ou=users,dc=puppetlabs,dc=com",
+                  "sshpublickey"=>
+        ["ssh-rsa AAAAB...1== user@somewhere",
+        "ssh-rsa AAAAB...2== user@somewhereelse"]}]
+
+        entries = Marshal.load(File.read("spec/fixtures/entries_multivalue.obj"))
+
+        l = PuppetX::LDAPquery.new(filter, attributes, base)
+        #expect(l).to receive(:get_config).and_return(real_conf)
```
Comment (danieldreier) [inline spec/unit/ldapquery_spec.rb]: dead code?

File: lib/puppet_x/ldapquery.rb (hunk 1 above)
Comment (xaque208) [inline lib/puppet_x/ldapquery.rb]: Ah, thank you.

File: spec/fixtures/entries_multivalue.obj (hunk 2 above)
Comment (xaque208) [inline spec/fixtures/entries_multivalue.obj]: I think that happens when you marshall.

**Comment by danieldreier** [comment]
+1, nice work. I'm always happy to see new projects with good test coverage.

**Pull request merged**